ts/tests/commands/precompile-test.ts

```diff
@@ -79,4 +79,25 @@ describe('Acceptance: ts:precompile command', function() {
         });
     });
   });
+
+  it('generates .d.ts files when addon and package names do not match', function() {
+    fs.ensureDirSync('addon-test-support');
+    fs.writeFileSync('addon-test-support/test-file.ts', `export const testString: string = 'hello';`);
+
+    const index = fs.readFileSync('index.js', { encoding: 'utf-8' })
+      .replace("name: require('./package').name", "name: 'my-addon'")
+      .replace("}", "  isDevelopingAddon: () => true,\n}");
```

Comment:
We _could_ go the fixtures route, but if we end up not relying on `isDevelopingAddon` (see comment above), maybe it would be simpler to replace the entire file?

```ts
fs.writeFileSync('index.js', `module.exports = { name: 'my-addon' };`);
```

Reply:
It looks like this is producing a syntax error (missing comma), which is probably why the test isn't passing? I'm surprised ember-cli doesn't complain more loudly, though.

```js
'use strict';

module.exports = {
  name: 'my-addon'
  isDevelopingAddon: () => true,
};
```

Reply:
Doh, that would do it, yep. Updating now.